### PR TITLE
Don't suppress exception chaining for optional dependencies

### PR DESCRIPTION
**Pull request by takluyver**
I've just been [helping someone](https://stackoverflow.com/questions/69352179/package-streamlit-app-and-run-executable-on-windows) for whom the helpful error message appears to have obscured the real cause of the problem. Specifically, Jinja2 is present, but one of *its* dependencies is not. So the error message saying "Missing optional dependency: 'Jinja2'..." is rather confusing.

Automatic exception chaining is really great for things like this - you can provide a more descriptive error message while still having the original exception details available for inspection. I'd argue that it should only be disabled (`raise ... from None`) if you're absolutely sure that there will be no useful information in the original exception.

### CODE REVIEW

**Review by mzeitlin11 — APPROVED** [review]
Thanks @takluyver, looks like a nice improvement! Can you please add a quick note in the `whatsnew` for 1.4 mentioning this improved error message? Also might be nice to have a test for this (if not too cumbersome)

**Comment by takluyver** [comment]
Thanks, :crossed_fingers: that test works. I'm editing from the web interface. :slightly_smiling_face:

File: doc/source/whatsnew/v1.4.0.rst [hunk 1]
```diff
@@ -142,7 +143,8 @@ These are bug fixes that might have notable behavior changes.
 Inconsistent date string parsing
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-The ``dayfirst`` option of :func:`to_datetime` isn't strict, and this can lead to surprising behaviour:
+The ``dayfirst`` option of :func:`to_datetime` isn't strict, and this can lead to sur
+ising behaviour:
```
Comment (mzeitlin11) [inline doc/source/whatsnew/v1.4.0.rst:147]: Looks like an accidental change?

Comment (takluyver) [reply doc/source/whatsnew/v1.4.0.rst]: It was, sorry about that. I've reverted it in a new commit; feel free to squash it when merging.

File: doc/source/whatsnew/v1.4.0.rst [hunk 2]
```diff
@@ -127,7 +127,8 @@ Other enhancements
 - :meth:`IntegerArray.all` , :meth:`IntegerArray.any`, :meth:`FloatingArray.any`, and :meth:`FloatingArray.all` use Kleene logic (:issue:`41967`)
 - Added support for nullable boolean and integer types in :meth:`DataFrame.to_stata`, :class:`~pandas.io.stata.StataWriter`, :class:`~pandas.io.stata.StataWriter117`, and :class:`~pandas.io.stata.StataWriterUTF8` (:issue:`40855`)
 - :meth:`DataFrame.__pos__`, :meth:`DataFrame.__neg__` now retain ``ExtensionDtype`` dtypes (:issue:`43883`)
-
+- The error raised when an optional dependency can't be imported now includes the original exception, for easier investigation.
```
Comment (lithomas1) [inline doc/source/whatsnew/v1.4.0.rst:130]:
```suggestion
- The error raised when an optional dependency can't be imported now includes the original exception, for easier investigation (:issue:`43882`)
```

**Comment by mzeitlin11** [comment]
close/reopen for CI

**Review by lithomas1 — APPROVED** [review]

**Comment by takluyver** [comment]
Thanks all :slightly_smiling_face:

**Pull request merged**